addons/new-osio-addon/src/main/java/io/fabric8/launcher/osio/jenkins/JenkinsPipelineRegistry.java

```diff
@@ -13,32 +33,91 @@
+    private static final String JENKINS_PIPELINE_REPO_REF = "master";
 
+    private static final Logger log = Logger.getLogger(JenkinsPipelineRegistry.class.getName());
 
     /**
      * Builds the registry index. Should be called only once
      */
     @PostConstruct
     public void index() {
-//        try {
-//            Path pipelinesPath = Paths.get(getClass().getResource("pipelines").toURI());
-//            System.out.println("PIPELINES: " + pipelinesPath);
-//        } catch (URISyntaxException e) {
-//            e.printStackTrace();
-//        }
+        try {
+            Path pipelinesPath = clonePipelinesLibrary(JENKINS_PIPELINE_REPO_URL, JENKINS_PIPELINE_REPO_REF);
+
+            Yaml yaml = new Yaml();
+            Map<String, JenkinsPipeline> pipes =  new HashMap<>();
+            Files.walkFileTree(pipelinesPath, new SimpleFileVisitor<Path>() {
+                @Override
+                public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                    String name = file.getFileName().toString();
+                    if (name.equals("metadata.yml") || name.equals("metadata.yaml")) {
+                        JenkinsPipeline p = readPipeline(file);
+                        if (p != null) {
+                            pipes.put(p.getId(), p);
+                        }
+                    }
+                    return super.visitFile(file, attrs);
+                }
+            });
+            pipelines = Collections.unmodifiableMap(pipes);
+        } catch (IOException e) {
+            log.log(Level.SEVERE, "Unable to index Jenkins pipelines", e);
+        }
+    }
+
+    private Path clonePipelinesLibrary(String repoUrl, String repoRef) throws IOException {
+        log.log(Level.INFO, "Indexing contents from {0} using {1} ref",
+                new Object[]{repoUrl, repoRef});
+        Path targetPath = Files.createTempDirectory("pipeline-library");
+        log.info("Created " + targetPath);
+        ProcessBuilder builder = new ProcessBuilder()
+                .command("git", "clone", repoUrl, "--branch", repoRef,
+                        "--recursive", "--depth=1", "--quiet",
+                        targetPath.toString())
+                .inheritIO();
+        log.info("Executing: " + builder.command().stream().collect(Collectors.joining(" ")));
+        try {
+            int exitCode = builder.start().waitFor();
+            assert exitCode == 0 : "Process returned exit code: " + exitCode;
+        } catch (InterruptedException e) {
+            log.log(Level.WARNING, "Interrupted indexing process");
+            throw new IOException("Interrupted", e);
+        }
+        return targetPath;
+    }
 
+    private JenkinsPipeline readPipeline(Path metadataFile) {
+        Yaml yaml = new Yaml();
+        try (BufferedReader reader = Files.newBufferedReader(metadataFile)) {
+            @SuppressWarnings("unchecked")
```

Comment:
Better move this annotation to the `readPipeline` method level

Reply:
Yup